### PR TITLE
Enable email on localhost by default

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

- Enables sending email by default on `localhost`.
- Deprecates `DEFAULT_FROM_EMAIL` in favor of `EMAIL_DEFAULT_FROM`.
- Introduces `EMAIL_ENABLED` env var to globally enable/disable email.

## Checklist

- [X] N/A. All querysets/queries filter by Organization, Team, and User (if this PR affects ANY querysets/queries).
- [X] Django backend tests (if this PR affects the backend).
- [X] N/A. Cypress end-to-end tests (if this PR affects the frontend).

### CODE REVIEW

**Comment by paolodamico** [comment]
@timgl can you help me test email delivery on `localhost`? can't run it properly on OS X and there also seems to be some port blocking interfering, think it should work with Docker?

**Review by macobo — APPROVED** [review]

**Comment by macobo** [comment]
Code change looks fine! Didn't test it out manually though :)

File: posthog/settings.py [hunk 1]
```diff
@@ -431,13 +431,14 @@ def print_warning(warning_lines: Sequence[str]):
...
 # Email
-EMAIL_HOST = os.environ.get("EMAIL_HOST")
+EMAIL_ENABLED = get_bool_from_env("EMAIL_ENABLED", True)
+EMAIL_HOST = os.environ.get("EMAIL_HOST", "localhost")
 EMAIL_PORT = os.environ.get("EMAIL_PORT", "25")
 EMAIL_HOST_USER = os.environ.get("EMAIL_HOST_USER")
 EMAIL_HOST_PASSWORD = os.environ.get("EMAIL_HOST_PASSWORD")
 EMAIL_USE_TLS = get_bool_from_env("EMAIL_USE_TLS", False)
 EMAIL_USE_SSL = get_bool_from_env("EMAIL_USE_SSL", False)
-DEFAULT_FROM_EMAIL = os.environ.get("DEFAULT_FROM_EMAIL", "hey@posthog.com")
```
Comment (timgl) [inline posthog/settings.py:440]: Will this change the default email for _all_ SMTP servers too?

**Review by timgl — COMMENTED** [review]
This doesn't work without some kind of bridge to sendmail [like this one](https://github.com/perenecabuto/django-sendmail-backend)

**Comment by paolodamico** [comment]
Decided to drop the defaulting to `sendmail` for now because it requires having both an extra agent and `sendmail` installed in the source machine. Additional context of issue on [Gitlab](https://gitlab.com/gitlab-org/omnibus-gitlab/-/issues/2630). We could potentially add it to the Docker image, but we would then have to consider a few edge cases (e.g. Heroku doesn't allow sending email from your instance).

**Pull request merged**